### PR TITLE
Make SimpleMap dropable

### PR DESCRIPTION
**Pull request by areshand**
### Description
make the generic simple map dropable

### Test Plan
cargo test -p framework

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2355)

### CODE REVIEW

File: aptos-move/framework/aptos-stdlib/sources/simple_map.move [hunk 1]
```diff
@@ -13,11 +13,11 @@ module aptos_std::simple_map {
     const EKEY_ALREADY_EXISTS: u64 = 0;
     const EKEY_NOT_FOUND: u64 = 1;
 
-    struct SimpleMap<Key: store, Value: store> has store {
+    struct SimpleMap<Key: store, Value: store> has drop, store {
```
Comment (movekevin) [inline aptos-move/framework/aptos-stdlib/sources/simple_map.move:16]: This wouldn't work though right? The Key and Value would need to have drop as well. But that's changing the constraint quite a bit.

Comment (wrwg) [reply aptos-move/framework/aptos-stdlib/sources/simple_map.move:16]: You need to remove the `:store` (not constraint) and the type parameters. Similar on the `Element` type. This wont compile. See also slack chat.

Comment (movekevin) [reply aptos-move/framework/aptos-stdlib/sources/simple_map.move:16]: Nvm we discussed offline and this would work 👍

**Review by movekevin — APPROVED** [review]

File: aptos-move/framework/aptos-stdlib/sources/simple_map.move (hunk 1 above)
Comment (areshand) [reply aptos-move/framework/aptos-stdlib/sources/simple_map.move:16]: @wrwg  the code compiles now. But I can remove the :store on key, value and element to make the simple_map more versatile.

**Pull request merged**